multi-agent/adapters/outbound/langfuse/langfuse_tracing_service.py

```
@@ -0,0 +1,183 @@
+        with self._client.start_as_current_observation(
+            as_type="span",
+            name=f"session:{session_id}",
+            input=metadata,
+            trace_context={"trace_id": trace_id},
+        ) as obs:
+            with propagate_attributes(
+                session_id=session_id,
+                user_id=user_id or None,
+                metadata=metadata,
+                trace_name=f"session:{session_id}",
+            ):
```

Ruff (0.15.21)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)